controller/api/destination/watcher/ip_watcher_test.go

```diff
@@ -0,0 +1,462 @@
+package watcher
+
+import (
+	"sort"
+	"testing"
+
+	"github.com/linkerd/linkerd2/controller/k8s"
+
+	logging "github.com/sirupsen/logrus"
+)
+
+func TestIPWatcher(t *testing.T) {
+	for _, tt := range []struct {
+		serviceType                      string
+		k8sConfigs                       []string
+		host                             string
+		port                             Port
+		expectedAddresses                []string
+		expectedNoEndpoints              bool
+		expectedNoEndpointsServiceExists bool
+		expectedError                    bool
+	}{
+		{
```

Comment:
These use cases seem identical(-ish) to those defined in [`endpoints_watcher_test.go`](https://github.com/linkerd/linkerd2/blob/005bc0ca8ffc45474b18d3c4be76c7804aba1acd/controller/api/destination/watcher/endpoints_watcher_test.go#L60-L378). Any reasons why we need to re-run them here? Do the behaviour of the `endpoints_watcher` changes after it's being wrapped?

Reply:
the way the lookups works is different, with lookups by ip instead of by service name.  I think it's worth testing these conditions.